content/docs/2/reference-iofogctl/reference-template-engine.md

```diff
@@ -0,0 +1,237 @@
+                    "key": "sharedToken",
+                    "value": "{{ self.microservices | where: \"name\", \"rulesengine\" | first | map: \"env\" | first | where: \"key\", \"sharedToken\" | first | map: \"value\" | first }}"
+                },
+                {
+                    "key": "http_proxy",
+                    "value": "{{ self.microservices | where: \"name\", \"rulesengine\" | first | map: \"env\" | first | where: \"key\" , \"http_proxy\" | first | map: \"value\" | first }}"
+                },
+                {
+                    "key": "https_proxy",
+                    "value": "{{ self.microservices | where: \"name\", \"rulesengine\" | first | map: \"env\" | first | where: \"key\" , \"http_proxy\" | first | map: \"value\" | first }}"
+                },
+                {
+                    "key": "rulesengineHOST",
+                    "value": "{%  assign curmsvc= self.microservices | where: \"name\", \"msvc-1\" | first %}{{ curmsvc | findAgent: iofogs | map: \"host\" }}"
+                },
+                {
+                    "key": "rulesenginePORT",
+                    "value": "{{ self.microservices | where: \"name\", \"rulesengine\" | first | map: \"ports\" | first | map: \"external\" | first }}"
+                },
+                {
+                    "key": "redisHost",
+                    "value": "{% assign redismsvc = microservices | where: \"name\", \"redistest\" | first %}{{ redismsvc | findAgent: iofogs | map: \"host\"}}:{{ redismsvc | map: \"ports\" | first | first |map: \"external\" | first }}"
+                },
+                {
+                    "key": "edgeResLiveness",
+                    "value": "{{ \"com.orange.smart-door\" | findEdgeResource: \"0.0.1\" | map: \"interface\" | map: \"endpoints\" | first  | where: \"name\", \"liveness\" | first | map: \"url\" }}"
+                },
+                {
+                    "key": "edgeResVersion",
+                    "value": "{{ \"com.orange.smart-door\" | findEdgeResource            | map: \"interface\" | map: \"endpoints\" | first  | where: \"name\", \"version\" | first | map: \"url\" }}"
+                }
+            ],
+            "cmd": []
+        }
+    ]
+}
+```
+
+The context for scripting contains the variables:
+
+| variable      | Description                                                                                                           |
+| ------------- | --------------------------------------------------------------------------------------------------------------------- |
+| self          | This is the current object. if deploying an application, this is the current description of the application.          |
+| microservices | This is the list of the microservices deployed on the controller.                                                     |
+| iofogs        | This is the list of agents deployed on the controller.                                                                |
```

Comment:
Don't use `iofogs` here, use `agent`. `iofogs` is overloaded and we intended to remove "fogs" from the codebase over time.

Reply:
ok i use that because this is the name of the API. I must change the docs and the implementation.

Reply:
agents is **OK** ?

Reply:
Done, Docs, src and tests.